core/src/main/java/org/apache/iceberg/util/ParallelIterable.java

```diff
@@ -20,65 +20,69 @@
 
 import java.io.Closeable;
 import java.io.IOException;
+import java.util.ArrayDeque;
+import java.util.Deque;
 import java.util.Iterator;
 import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.concurrent.Callable;
 import java.util.concurrent.ConcurrentLinkedQueue;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Future;
-import org.apache.iceberg.exceptions.RuntimeIOException;
+import java.util.concurrent.atomic.AtomicBoolean;
 import org.apache.iceberg.io.CloseableGroup;
 import org.apache.iceberg.io.CloseableIterable;
 import org.apache.iceberg.io.CloseableIterator;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
+import org.apache.iceberg.relocated.com.google.common.io.Closer;
 
 public class ParallelIterable<T> extends CloseableGroup implements CloseableIterable<T> {
+
+  private static final int DEFAULT_MAX_QUEUE_SIZE = 10_000;
```

Comment:
How do we know that this is a reasonable size? I don't think that we want to choose a size that is too small or else we could introduce slower planning because of tasks yielding. Assuming that a DataFile or DeleteFile is about 500 bytes, a 10k limit uses only 4.7 MB of memory. I think we could bump that up without a problem.

Reply:
I think it makes sense to resume when about half-way through the queue. What do you think?

Reply:
done!

Reply:
per offline suggestion switched to 30k and added this comment:

```
  // Logic behind default value: ParallelIterable is often used for file planning.
  // Assuming that a DataFile or DeleteFile is about 500 bytes, a 30k limit uses 14.3 MB of memory.
  private static final int DEFAULT_MAX_QUEUE_SIZE = 30_000;
```